File: 0282-18-3208-alternating-groups-ii/0282-18-3208-alternating-groups-ii.java
class Solution {
    public int numberOfAlternatingGroups(int[] colors, int k) {
        int n = colors.length;
        int cnt = 1;
        int ans = 0;
        for (int i = 1; i < n + k - 1; i++) {
            int idx = (i >= n) ? i - n : i;
            int prevIdx = (i - 1 >= n) ? (i - 1) - n : (i - 1);
            cnt = (colors[idx] != colors[prevIdx]) ? cnt + 1 : 1;
            ans += (cnt >= k) ? 1 : 0;
        }
        return ans;
    }
}

File: 0282-18-3208-alternating-groups-ii/0282-18-3208-alternating-groups-ii.py
class Solution:
    def numberOfAlternatingGroups(self, colors: list[int], k: int) -> int:
        n = len(colors)
        cnt = 1
        ans = 0
        for i in range(1, n + k - 1):
            idx = i if i < n else i - n
            prev_idx = (i - 1) if (i - 1) < n else (i - 1) - n
            cnt = cnt + 1 if colors[idx] != colors[prev_idx] else 1
            ans += 1 if cnt >= k else 0
        return ans